netlify.toml

```
@@ -7,13 +7,17 @@
   RUBY_VERSION = "2.5.4"
   RELEASE_VERSION = "v3.14"
 
+# One jekyll build here generates the website which is served at /archive/v3.14/ other build generates only manifests
+# served on /v3.14/ path, order of the build matters since destination folders are cleaned on each builds
+# 404 page moved to root of the site so that netlify shows it automatically
 [context.branch-deploy]
   command = '''
   cp netlify/Gemfile Gemfile && bundle install
   make bin/helm
   export PATH=$PATH:$(pwd)/bin
-  jekyll build --config _config.yml,$(pwd)/netlify/_config_noindex.yml --baseurl /archive/$RELEASE_VERSION
   jekyll build --config _config.yml,$(pwd)/netlify/_config_noindex.yml,$(pwd)/netlify/_manifests_only.yml --baseurl /$RELEASE_VERSION --destination _site/$RELEASE_VERSION
+  jekyll build --config _config.yml,$(pwd)/netlify/_config_noindex.yml --baseurl /archive/$RELEASE_VERSION --destination _site/archive/$RELEASE_VERSION
+  mv _site/archive/$RELEASE_VERSION/404.html _site/404.html
```

This is extremely nitpicky, and possibly just a matter of taste, but I think it'd be clearer if we comment each line
```suggestion
  # generate the website which is served at /archive/<version>/ 
  jekyll build --config _config.yml,$(pwd)/netlify/_config_noindex.yml,$(pwd)/netlify/_manifests_only.yml --baseurl /$RELEASE_VERSION --destination _site/$RELEASE_VERSION
  # generate only manifests served at /<version>
  # note: this must be done second since destination folders are cleaned on each builds
  jekyll build --config _config.yml,$(pwd)/netlify/_config_noindex.yml --baseurl /archive/$RELEASE_VERSION --destination _site/archive/$RELEASE_VERSION
  # move 404 page to root of the site so that netlify shows it automatically
  mv _site/archive/$RELEASE_VERSION/404.html _site/404.html
```

---

makes sense, I'll commit this and update other comments